pkg/kube/install-etcdctl.sh

```diff
@@ -4,6 +4,20 @@
 # SPDX-License-Identifier: Apache-2.0
 
 ETCDCTL_VERSION=v3.5.5
-/usr/bin/wget https://github.com/etcd-io/etcd/releases/download/${ETCDCTL_VERSION}/etcd-${ETCDCTL_VERSION}-linux-amd64.tar.gz
-tar -zxv --strip-components=1 -C /usr/local/bin  < ./etcd-${ETCDCTL_VERSION}-linux-amd64.tar.gz
-rm ./etcd-${ETCDCTL_VERSION}-linux-amd64.tar.gz
+# Detect architecture
+ARCH=$(uname -m)
+if [ "$ARCH" = "x86_64" ]; then
+    ARCH="amd64"
+elif [ "$ARCH" = "aarch64" ]; then
+    ARCH="arm64"
+else
+    echo "Unsupported architecture: $ARCH"
+    exit 1
+fi
+
+# Download the appropriate etcd binary
+/usr/bin/wget https://github.com/etcd-io/etcd/releases/download/${ETCDCTL_VERSION}/etcd-${ETCDCTL_VERSION}-linux-${ARCH}.tar.gz
```

Comment:
Can you also download virtctl here ?  Just move from pkg/kube/Dockerfile to here. That saves 30MB or so in image.
You can also do as separate PR, if you prefer.

Otherwise LGTM.

Reply:
virtctl is used by 'remote console', this above is manually done. we can build in some logic to dynamically downloading virtctl when it is needed.